#23 - Add source comments from file descriptor

n3rdyme opened the pull request:
Added comments from fileDescriptor.sourceCodeInfo to enums, message interfaces, and services.

Should resolve issue 23
https://github.com/stephenh/ts-proto/issues/23

stephenh commented on src/sourceInfo.ts:
```diff
@@ -0,0 +1,88 @@
+    },
+}
+
+export interface SourceDescription { 
```
Can you add a link to the docs (assuming they exist, maybe the protobuf declaration file?) of what these fields are?

n3rdyme replied:
@stephenh Done.

stephenh commented on src/sourceInfo.ts:
```diff
@@ -0,0 +1,88 @@
+export type FieldID = number;
+
+export const Fields = {
+    file: {
```
I don't have it setup to auto-run, but there is a `.prettierrc` checked in; can you run prettier on this?

n3rdyme replied:
@stephenh Done.

stephenh commented on src/sourceInfo.ts:
```diff
@@ -0,0 +1,88 @@
+};
+
+export type SourceInfoMap = {
+    [key: string]: SourceDescription
```
Maybe add an in-source comment about what some example keys are?

n3rdyme replied:
@stephenh Done.

stephenh commented on src/sourceInfo.ts:
```diff
@@ -0,0 +1,88 @@
+        return this.sourceCode[`${type}.${index}`] || new EmptyDescription();
+    }
+
+    open(type: FieldID, index: number): SourceInfo {
```
What's the different between `lookup` and `open`? It looks like open...finds multiple results? Add in-source method comments for each?

n3rdyme replied:
@stephenh Added some comments... generally lookup is get one thing, open is open the type so I can lookup more relative to that type. Another way to say it, Message has nested types, so use `open`, for everything else you can just use `lookup`.

n3rdyme commented:
@stephenh 
I added the requested comments in simple.proto and cleaned up the conditional logic a bit. I assume you did not want me to regenerate all the binary and source files on branch. It seems like that might make merging difficult, so I assume you are keeping PR branches clean?

Otherwise, I think I covered all the asks above. Let me know if you need any additional changes.

stephenh commented:
Looks great, thanks! I do actually check in the generated source files; I know in downstream / real projects that is an anti-pattern but given ts-proto's job is producing that output, I basically treat it as an mini-integration test to see how the output has changed in the PRs/diffs/etc. I'll go ahead and do a follow up commit with those changes now.
